pkg/ruler/ruler.go

```diff
@@ -59,6 +60,8 @@ var (
 type Config struct {
 	// This is used for template expansion in alerts; must be a valid URL.
 	ExternalURL flagext.URLValue `yaml:"external_url"`
+	// TLS parameters for the GRPC Client
+	ClientTLSConfig tls_cfg.ClientConfig `yaml:",inline"`
```

Comment:
The YAML config is inline, while the CLI flag is prefixed by `-ruler.client` (ie. `-ruler.client.tls-cert-path`). This is a config inconsistency that we should fix. I think having the struct inline here is quite confusing (for which specific client are we configuring TLS?), so I would suggest to not inline the YAML struct here, so:

```suggestion
	ClientTLSConfig tls_cfg.ClientConfig `yaml:"ruler_client"`
```

I've named it `ruler_client` (and CLI flag prefix should be reflected as well) because it's the client that a ruler uses to connect to other rulers. The ruler can also connect to other endpoints (ie. alert manager) so it's quite important to avoid ambiguities here.